Cargo.toml

```diff
@@ -41,7 +41,7 @@ core2 = { version = "0.3.0", optional = true, default-features = false }
 
 base64-compat = { version = "1.0.0", optional = true }
 bitcoinconsensus = { version = "0.19.0-3", optional = true }
-serde = { version = "1", features = [ "derive" ], optional = true }
+serde = { version = "1", default-features = false, features = [ "derive" ], optional = true }
```

Comment:
This will result in the use of non-std version of serde even if both `serde` and `std` features are used for `rust-bitcoin` compillation

Reply:
Yes, I am not aware of a great way around this. In fact, as it stands today, a no-std build will still compile `serde/std`, which seems equally misleading. Attempting to build for a target which does not support std results in compile failures.

A possible compromise would be to re-export the `serde/std` feature, so that a dependent project can enable it if desired.

I'm also open to suggestions, if you have other ideas how to allow serde in pure no-std builds. Or perhaps drop a comment in the parent issue: https://github.com/rust-bitcoin/rust-bitcoin/issues/904

Reply:
Related to this question, if compiling for `no-std`, wouldn't it be best to ensure all dependencies are also compiling for `no-std`? I thought this was the case until I found out the hard way that Serde is still compiling with std enabled. At best its confusing. At worst it could lead to unintended use of shared system libraries.

Reply:
Can't we just add `serde/std` to the `std = ["secp256k1/std", "bitcoin_hashes/std", "bech32/std"]` line?

Reply:
Ah, right, no, because that would pull `serde` in. Never mind. Cargo..

Reply:
That was my original thought too, but Cargo... :sob:

Surely this isn't the first time a project has needed conditional features in the manifest? I spent some time searching for better workarounds, but so far no luck. I'm all ears if someone has a better way to do this.

Reply:
We have run into this probably dozens of times. It is a known issue but there is still no fix.

We should probably just add the extra `serde-std` feature.

Reply:
Yes, this is an old issue and there are exactly two solutions:

* add `serde-std` feature
* bump MSRV to 1.60 - the one to be released next

Reply:
Added `serde-std` and confirmed builds+tests pass on stable, nightly and 1.29.0.